app/src/main/java/com/example/sopt_1/SignUpActivity.kt

```diff
@@ -0,0 +1,87 @@
+package com.example.sopt_1
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import com.example.sopt_1.databinding.ActivitySignUpBinding
+
+class SignUpActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySignUpBinding
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignUpBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        Log.d(CURRENT_ACTIVITY, "Called onCreate")
+
+        signUpButtonClickEvent()
+
+    }
+
+    private fun signUpButtonClickEvent() {
+        binding.btnSignUp.setOnClickListener {
+
+            val userName = binding.etSignupName.text
+            val userId = binding.etSignupId.text
+            val userPw = binding.etSignupPw.text
+            if (userName.isNullOrBlank() || userId.isNullOrBlank() || userPw.isNullOrBlank()) {
+                Toast.makeText(this@SignUpActivity, "빈 칸이 있는지 확인해주세요", Toast.LENGTH_SHORT)
+                    .show()
+            } else {
+                //초기 SignUpActivity로 돌아갈 수 있도록 종료
+                // 종료 전 putExtra를 이용해 모든 값을 intent에 넣어 전달
+                val intent = Intent(this@SignUpActivity, HomeActivity::class.java)
+                intent.putExtra("userName", userName.toString())
+                intent.putExtra("userId", userId.toString())
+                intent.putExtra("userPw", userPw.toString())
```

Comment:
이렇게도 쓸 수 있지만, 아래 코드와 같이 바꿔 쓸 수도 있으니 참고하시면 좋을 것 같아요 :)
```suggestion
                intent.putExtra("userName", userName.toString())
                .putExtra("userId", userId.toString())
                .putExtra("userPw", userPw.toString())
```